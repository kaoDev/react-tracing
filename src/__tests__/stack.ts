// @flow
import Stack from "../stack";

describe("stack", () => {
	it("should throw an error if is no element to peek on", () => {
		const s = new Stack();
		expect(() => {
			s.peek();
		}).toThrowErrorMatchingSnapshot();
	});

	it("should return the element on top of the stack if there is an element", () => {
		const s = new Stack();
		s.push(3);
		expect(s.peek()).toBe(3);
	});

	it("should remove the element on top", () => {
		const s = new Stack();
		s.push(1);
		s.push(2);
		s.push(3);

		expect(s.pop()).toBe(3);
		expect(s.peek()).toBe(2);
	});

	it("should remove an element given", () => {
		const s = new Stack();
		s.push(1);
		s.push(2);
		s.push(3);

		expect(s.length).toBe(3);
		s.remove(2);
		expect(s.length).toBe(2);
		expect(s.peek()).toBe(3);
	});
});
