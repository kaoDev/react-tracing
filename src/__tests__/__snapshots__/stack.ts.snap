// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stack should throw an error if is no element to peek on 1`] = `"No items in stack"`;
